keanu-project/src/main/java/io/improbable/keanu/plating/PlateBuilder.java

```diff
@@ -118,44 +144,73 @@ public FromDataFactory withFactory(BiConsumer<Plate, T> factory) {
         private BiConsumer<Plate, T> factory;
         private PlateData<T> data;
         private int size;
+        private final VertexDictionary initialState;
 
-        private FromDataFactory(BiConsumer<Plate, T> factory, PlateData<T> data, int size) {
+        private FromDataFactory(BiConsumer<Plate, T> factory, PlateData<T> data, int size, VertexDictionary initialState) {
             this.factory = factory;
             this.data = data;
             this.size = size;
+            this.initialState = initialState;
         }
 
-        public Plates build() {
+        public Plates build() throws VertexLabelException {
             Plates plates = new Plates(this.size);
             Iterator<T> iter = data.getIterator();
+            VertexDictionary previousPlate = initialState;
             while (iter.hasNext()) {
                 Plate plate = new Plate();
                 factory.accept(plate, iter.next());
+                connectProxyVariables(previousPlate, plate, proxyMapping);
                 plates.add(plate);
+                previousPlate = plate;
             }
             return plates;
         }
     }
 
+    private void connectProxyVariables(VertexDictionary candidateVertices, Plate plate, Map<VertexLabel, VertexLabel> proxyMapping) throws VertexLabelException {
```

Comment:
I know these make use of `ProxyVertex` but I think a name that describes it's a map between vertices representing the same thing just at different timesteps would make it a lot clearer what's going on. `connectTimesteppedVariables`? And then we could have `withTimestepMapping` on the builder

Reply:
It's for loops as well as time series, so `timestep` isn't quite the right name to use. Maybe `withTransitionMapping`? In time series we talk about the transition model (c.f. Charlie's talk yesterday) and that's more-or-less applicable to loop iterations too.

Reply:
`withTransitionMapping` sounds good to me